twilio-labs/paste
# chore: add create-docs script and templates

## In this PR

- Adds a new script `yarn create:docs` to the `package.json` file to create a docs directory and necessary files
- Adds a new generator to the `plopfile.js` (`create-docs`)
- Adds 3 new plop template files: `docs-index.hbs`, `docs-api.hbs`, and `docs-changelog.hbs`
- Updates the `packages/paste-website/src/pages/components/account-switcher/index.mdx` file to pull the package description from the package itself, meaning we now have a single source of truth for the package description (instead of 4). This file demonstrates the change needed to streamline the package description for all other packages. When anyone touches another website page (or just has free time/space in their PR), let's update the rest of the component website files with this same change.

## Limitations
There are certain places where I've added `TODO:`s to the templates. They will require attention and slight changes after the script is run. (If you have the "TODO Highlight" VS Code extension installed, the `TODO:`s will all be highlighted and hard to miss.)
- plopjs doesn't handle `titleCase` well, so in order for the website to build, there are a few places the case will need to be adjusted, or a space will need to be added ([there's an open issue for this](https://github.com/plopjs/plop/issues/415), I added a comment to it)
- The way you use variables in plop is with double curly brackets (e.g. `{{component-name}}`). That prevents me from being able to add curly brackets to the plop templates without causing plop to fail. Because our LivePreview also uses double curly bracket notation (e.g. `scope={{ComponentName}}`), the brackets will have to be added after the script is run
- Storybook file urls will need to be checked and possibly updated once the story urls are finalized

## How to test
If you want to give the new script a try to make sure it's working correctly, checkout this branch and delete an existing component's directory within `packages/paste-website/src/pages/components` and run the script for that component. Make sure the website builds after running the script, and try giving the plop prompt different variations of the component name ("progress steps", "ProgressSteps", "progress-steps", etc).
